### PR TITLE
Bump Python and RAPIDS versions

### PR DESCRIPTION
**Pull request by ryantwolf**
NeMo only supports `python>=3.10` so we need to bump our version too. Only bumping the Python version gives this error during installation:
```
cugraph 23.12.0 requires dask-cuda==23.12.*, but you have dask-cuda 23.10.0 which is incompatible.
cugraph-service-server 23.12.0 requires dask-cuda==23.12.*, but you have dask-cuda 23.10.0 which is incompatible.
cuml 23.12.0 requires dask-cuda==23.12.*, but you have dask-cuda 23.10.0 which is incompatible.
raft-dask 23.12.0 requires dask-cuda==23.12.*, but you have dask-cuda 23.10.0 which is incompatible.
rapids-dask-dependency 23.12.1 requires dask==2023.11.0, but you have dask 2023.9.2 which is incompatible.
rapids-dask-dependency 23.12.1 requires distributed==2023.11.0, but you have distributed 2023.9.2 which is incompatible.
```

By bumping the RAPIDS packages too, this error goes away. GPU Unit tests pass as well. Addresses #17

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -55,10 +55,10 @@
...
         "beautifulsoup4",
         "mwparserfromhell @ git+https://github.com/earwig/mwparserfromhell.git@0f89f44",
-        "cudf-cu12==23.10.*",
-        "dask-cudf-cu12==23.10.*",
-        "cugraph-cu12==23.10.*",
-        "dask-cuda==23.10.*",
+        "cudf-cu12==23.12.*",
+        "dask-cudf-cu12==23.12.*",
+        "cugraph-cu12==23.12.*",
+        "dask-cuda==23.12.*",
```
Comment (VibhuJawa) [inline setup.py:61]: I think we want to bump this way up to `24.02` which is the latest stable RAPIDS release.  (See release blog [link](https://medium.com/rapids-ai/rapids-24-02-release-43ad88db0627))

```suggestion
        "cudf-cu12==24.02.*",
        "dask-cudf-cu12==24.02.*",
        "cugraph-cu12==24.02.*",
        "dask-cuda==24.02.*",
```

**Review by VibhuJawa — COMMENTED** [review]
We should bump to `24.02` , latest stable rapids release. @ayushdg , any objections ?

File: setup.py (hunk 1 above)
Comment (ryantwolf) [reply setup.py:61]: Is there a clean way to pull of a lower bound on the RAPIDS libraries so we don't have to manually update the versions? Not sure if that's good practice though. In the "Remove wildcard suffix" commit I tried to do it but got these errors:
```
cugraph 23.12.0 requires dask-cuda==23.12.*, but you have dask-cuda 24.2.0 which is incompatible.
cugraph 23.12.0 requires rapids-dask-dependency==23.12.*, but you have rapids-dask-dependency 24.2.0 which is incompatible.
cugraph-service-server 23.12.0 requires dask-cuda==23.12.*, but you have dask-cuda 24.2.0 which is incompatible.
cugraph-service-server 23.12.0 requires rapids-dask-dependency==23.12.*, but you have rapids-dask-dependency 24.2.0 which is incompatible.
cuml 23.12.0 requires dask-cuda==23.12.*, but you have dask-cuda 24.2.0 which is incompatible.
cuml 23.12.0 requires rapids-dask-dependency==23.12.*, but you have rapids-dask-dependency 24.2.0 which is incompatible.
dask-cudf 23.12.0 requires rapids-dask-dependency==23.12.*, but you have rapids-dask-dependency 24.2.0 which is incompatible.
raft-dask 23.12.0 requires dask-cuda==23.12.*, but you have dask-cuda 24.2.0 which is incompatible.
raft-dask 23.12.0 requires rapids-dask-dependency==23.12.*, but you have rapids-dask-dependency 24.2.0 which is incompatible
```

Comment (VibhuJawa) [reply setup.py:61]: Unsure why your lower bound commit failed, trying again locally to see if i can repro.

Comment (VibhuJawa) [reply setup.py:61]: FWIW, below seems to be working with base environment `python=3.10`

Currently it seems like for RAPIDS pip installation  is supported in  Python 3.9 and 3.10. 
The nightly builds also support 3.11, so in the next release that should work too. 

image


```diff
diff --git a/setup.py b/setup.py
index ae51575..0f4fc95 100644
--- a/setup.py
+++ b/setup.py
@@ -36,7 +36,7 @@ setup(
         "Programming Language :: Python :: 3",
     ],
     packages=find_packages(),
-    python_requires=">=3.7",
+    python_requires=">=3.10",
     install_requires=[
         "dask[complete]>=2021.7.1",
         "distributed>=2021.7.1",
@@ -55,15 +55,16 @@ setup(
         "comment_parser",
         "beautifulsoup4",
         "mwparserfromhell @ git+https://github.com/earwig/mwparserfromhell.git@0f89f44",
-        "cudf-cu12==23.10.*",
-        "dask-cudf-cu12==23.10.*",
-        "cugraph-cu12==23.10.*",
-        "dask-cuda==23.10.*",
+        "cudf-cu12>=24.2",
+        "dask-cudf-cu12>=24.2",
+        "cugraph-cu12>=24.2",
+        "dask-cuda>=24.2",
         "spacy>=3.6.0, <4.0.0",
         "presidio-analyzer==2.2.351",
         "presidio-anonymizer==2.2.351",
         "usaddress==0.5.10",
         "nemo_toolkit[nlp]>=1.23.0",
+       
     ],
     entry_points={
         "console_scripts": [
```

Comment (VibhuJawa) [reply setup.py:61]: I think we should set `python_requires=3.10` and then once rapids releases (which should be on  Wed, Apr 12, 2023)we can bump it to >=.  
	 
https://github.com/rapidsai/build-planning/issues/3

**Review by VibhuJawa — APPROVED** [review]
LGTM

**Review by ayushdg — APPROVED** [review]
Lgtm! Thanks for the changes.

**Comment by Maghoumi** [comment]
@ryantwolf I suggest updating README.md and explicitly mentioning the minimum required Python version.

**Pull request merged**